### DISCUSSION TITLE
Directory Sharing not working, though  SPICE Guest Tools are installed successfully

### DISCUSSION THREAD

**Discussion by MuhammadAliYousaf** [utmapp/UTM#3583] [UTM Help (macOS)]
I did install SPICE Guest Tools, but Directory sharing is not working. Clipboard copy past and internet is working. I tried multiple times after creating new VMs but it's not working.

Following is the screen shot of my VM's setting.
<img width="852" alt="Screenshot 2022-02-05 at 1 25 49 PM" src="https://user-images.githubusercontent.com/2880208/152634485-e4621f9e-9539-4f1a-988c-ba2f6305c7c3.png">

**Comment by adespoton** [comment]
Which directory are you sharing?  Can you share the debug log? Which version of UTM are you using?  Which host OS are you using?  Which architecture (x86 or M1, Mac or mobile) are you using?

**Reply by micaelmbagira** [reply]
@adespoton Same issue here, I am using UTM 3.0.4 (46) on Mac M1 12.1. I checked in `/media` but the directory's not there.

**Reply by MuhammadAliYousaf** [reply]
@adespoton 
I am sharing 'Downloads' folder, and I also tried to share other folders as well.
UTM version is 3.0.4 (46)
Installed windows 10, my machine is Apple M1 Pro, 16GB and 500GB

Please tell me the path from where I can get the Logs, I will share. Thanks

**Reply by Rembrandt001** [reply]
Have you installed Office? Office 365 kills sharing 3.04 also 3.10

**Reply by MuhammadAliYousaf** [reply]
No. In fact, I did not install any software.

**Reply by emailmichaeldinh** [reply]
> Have you installed Office? Office 365 kills sharing 3.04 also 3.10

That's what I am experiencing as well after I install office 365. Why is this the case and do you have a solution/workaround? 
The reason for installing UTM is to be able to use Excel on windows as it is more powerful.

Thank in advance

**Comment by Scaramanga89** [comment]
Same here, the shared folder shows the folder but the items inside are all greyed out. Also, is there no ability to drag and drop from Mac to the virtual machine? I can share them using the virtual drive on the Mac, but it includes all the hidden folders/extensions/config files as well so it's a bit of a mess.

**Reply by numetalkid2003** [reply]
Hi, I was also under the impression that I could just haul files over from the 'shared directory' screen-- but as it turns out this is just for selecting which folder's going to be shared. How you actually access the files is, well in the case of Windows, you go to 'This PC' in the VM and there's a Z: drive called st like 'davRoot' containing whatever you set as your shared folder.

**Comment by HappySpaceInvader** [comment]
Same problem for me...

- Host:  UTM Version 3.2.4 (58) on M1 Mac running macOS Monterey 12.5 (21G72)
- Guest:  Windows 7 Ultimate with Spice Guest Tools installed

Attempted to share ~/Downloads, but nothing appears in My Computer (expecting a new drive letter... right?)

And the shared clipboard doesn't work either.

The problem with shared folders is that although "SPICE Guest Tools" installs a "Spice WebDAV Proxy" service on the guest... there's something else we need to install (on the host?  on the guest?).

**Reply by 20har** [reply]
Hey did you find a solution for this one?

**Comment by grupot** [comment]
No he podido compartir directorio, en Mac M1, alguna solución?

**Comment by cnfanhua** [comment]
my machine also does't work(Apple M1 Pro, 32GB and 1TB).

UTM version: 3.2.4 (58)

Installed windows 11 22H2(ARM)

**Comment by Stean** [comment]
I had the same issue (at least it looks to me that we're talking about the same issue) and for me the workaround at https://github.com/utmapp/UTM/discussions/3755#discussioncomment-2487252 worked

**Reply by CollinMcKinney** [reply]
This worked for me with a few caveats.

1: When opening the folder with Dolphin File Manager it takes a long time to download the shared files if there are a lot of them or if it's a large size, and trying to interact with the file manager before loading is down caused it to freeze.

2. Since it was slow with Dolphin I decided to try switching to Nautilus. Nautilus had a similar issue but at least it didn't freeze, just took a long time to load the files.

3. Nautilus displayed the Spice Client Folder as being mounted, so i clicked that, and all the files are there instantly, usable, and it doesn't freeze up. So I went ahead and reversed all the steps shown in [#3755](https://github.com/utmapp/UTM/discussions/3755#discussioncomment-2487252)

4. Considering going back to Dolphin and using KDE Connect, instead of using UTM's sharing option with Spice.

**Comment by kimjansheden** [comment]
Running Windows 11 on a Mac M1 Pro and I had the same problem – shared drive didn't show in Windows file explorer. I had made a folder in Finder that I called "Shared" but I couldn't see it in Windows.

I found a very simple solution. The folder I tried to share was empty!!! As soon as I added something to it, it showed in Z:\! That worked for me at least, very simple 😂

**Reply by devonairedesigns** [reply]
Bro, this IS THE EXACT SOLUTION! Thank you!!!!!

**Comment by Kaiviti57** [comment]
It looks like kimjansheden has tumbled on the answer.  I have been trying to use the "shared Folder" icon in the top right, as I am sure most others are doing.  The shared folder is actually Z drive.  Use Windows Explorer, go to Network Drive (Z:) and your shared files are accessible from there.

**Comment by KPAdhikari** [comment]
I also thought I didn't get the Shared folder (was expecting the same name as on the Macbook/host side), but now I realized the shared folder was already showing up although I didn't notice that until I added a test file into it and looked around here and there. The name that showed up on the Windows (Guest) side was "DavWWWRoot (\\localhost@9843) (Z:)".

**Comment by jbergandino** [comment]
After installing the Spice tools (`sudo apt install spice-vdagent spice-webdavd`), I needed to shut down the VM and manually change the "Directory Share Mode" from "VirtFS" to "SPICE WebDAV" in the UTM VM settings.

Settings -> Sharing -> Directory Share Mode -> SPICE WebDAV

**Reply by knealking** [reply]
Here's a step by step instruction for the UTM setup
https://support.cci.drexel.edu/software-courses/utm-virtualization-m1-macs/utm-sharing-folder-linux-guest/#:~:text=First%2C%20when%20you%20launch%20UTM,you'd%20like%20to%20share.

Looks like it takes a couple of error messages for this work. Also needs an internet connection to be able to share files between your host machine and virtual machine for some reason.

**Comment by onaparrado** [comment]
I cannot get it to work with the solutions suggested in previous answers. 

- My setup: MacOS Ventura on Apple Silicon, UTM v4.2.5 (81), Windows 11 build 22621 as guest.
- UTM detects the shared folder (the icon is not greyed out).

<img width="1051" alt="image" src="https://github.com/utmapp/UTM/assets/28660385/37539380-3863-402a-888f-a3f0bfad18a7">

- The shared folder contains files, and has been successfully shared before on an Ubuntu VM.
- Spice guest tools are installed, and the SPICE VDAgent service is running.

<img width="802" alt="image" src="https://github.com/utmapp/UTM/assets/28660385/fe102ea4-099e-40e6-9aae-aa5b37c69db1">

- No Z drive was detected, also there is nothing under Network.

<img width="784" alt="image" src="https://github.com/utmapp/UTM/assets/28660385/21fd8208-f711-4673-93f0-338d1e367229">

- The network is set up as discoverable (private network).

<img width="784" alt="image" src="https://github.com/utmapp/UTM/assets/28660385/0ce63644-60eb-4341-a893-a313a8fd5161">


- I cannot map a network drive to \localhost@9843.

<img width="518" alt="image" src="https://github.com/utmapp/UTM/assets/28660385/be1a5b4c-9e88-4373-a921-69eae66c00bd">

- Office is not installed yet.
- The VM is connected to the internet, and I can see its screen from the host.
- SPICE webdav is selected in the VM config, and clipboard sharing works.

Any help would be appreciated!

**Reply by onaparrado** [reply]
Following the response of @basicallybene I've installed spice-guest-tools, but I cannot reinstall the WebDAV daemon as [there is no ARM version](https://www.spice-space.org/download/windows/spice-webdavd/) listed. Furthermore, the last build listed is v2.4, but UTM installed ARM v2.5 on my guest W11, so maybe it's their own build? Also, the [link to the repository](https://git.gnome.org/browse/phodav/tree/spice) is broken, as it targets to a branch that does not exist.

EDIT: UTM uses its own ARM port: https://github.com/utmapp/spice-webdavd-arm64. But the last commit is 3 years old, maybe we need v3.0 instead of 2.5 for W11.

**Comment by tiubenedict** [comment]
On Windows, I reinstalled both spice-guest-tools and spice WebDAV daemon from spice-space.org (because UTM's spice tools didnt work).

After a reboot, I used the "map network drive" feature and put \\\\localhost@9843, and it finally worked!

Edit:
Host: UTM Version 4.2.5 (81) on Intel Mac running macOS Monterey 12.6.3
Guest: Windows 7 Pro x64 with UTM Spice Tools as well as Spice-Space.org's spice-guest-tools and spice WebDAV

**Reply by tiubenedict** [reply]
After another reboot, Explorer now shows both the mapped drive (added by me) as well as a new Network Drive of the same shared folder. Not sure why it only kicked in now. Will upload a screenshot later.

**Reply by garfieldnate** [reply]
Thank you! I also re-installed the guest tools from the spice-space website, but I suspect that the key is really that the webDAV stuff is not included in the guest tools. The [UTM docs](https://docs.getutm.app/guest-support/sharing/directory/#selecting-directory-to-share) currently state that webDAV is the only usable option for Mac host, Windows guest, so I think that UTM should be including the webDAV installer on the built-in CD image for installing guest tools.

**Comment by atomlab** [comment]
Same problem with _Arch Linux_ as guest and _macOS Ventura 13.4.1 M1_ as host. The clipboard does not work between the host system and the guest system. Clipboard sharing is enabled in the UTM machine settings.

![Screenshot 2023-06-30 at 21 11 26](https://github.com/utmapp/UTM/assets/1222449/553dda49-8e09-41af-8580-61adb00bff4a)


All agents are installed in the guest system according to [documentation](https://docs.getutm.app/guest-support/linux/).

```
pacman -S qemu-guest-agent
pacman -S spice-vdagent
pacman -S phodav
```

**Comment by Kaiviti57** [comment]
Yes I did solve the issue.  It was a fair while back and I don’t use UTM anymore.  Sorry, I can’t help any further.

Capt. IVAN MAESEPP, Reef Pilot
251 WARNER ROAD | WRIGHTS CREEK, QLD. 4869
Mobile: 0428 020 559
Send me an Email <mailto: ***@***.***>
 <https://apextraderfunding.com/member/aff/go/kaiviti57?i=1>

**Comment by jmrodriguez** [comment]
This article helped me with this problem

https://jun1okamura.medium.com/install-ubuntu-on-mac-m1-powered-by-utm-499aba3ba7e9

M1 mac with osX 12.7.1 running ubuntu 22.04 on UTM 4.4.5

**Reply by cheshi-mantu** [reply]
as I see people are talking about Win on M1 not ubuntu. These two have different options to share folders – VirtFs mentioned in the article wil work for Linux, still it's completely useless for Windows guest

**Comment by cheshi-mantu** [comment]
UTM 4.4.4
Host: Apple M1 Pro
Guest: Win11 Home
WebDav drive does not appear on guest OS
reinstallation of Guest tools didn't help

**Comment by BlackthornYugen** [comment]
This was driving me crazy for a few days. I've re-installed the tools and restarted a bunch, but it wasn't until I actually shutdown and started it clean that the share started working. Credit to [Individual-Lime-6706 on reddit](https://www.reddit.com/r/MacOS/comments/sw9hdh/comment/kt1qcov/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button).

**Comment by SturmUDrang** [comment]
For reference. For me what worked was changing the directory to be shared. At first I had selected the Downloads folder, but apparently it had too many files. After that I have changed to a folder with less contents and now it works.

**Reply by cheshi-mantu** [reply]
this actually helped, cheers.
I tried to map Downloads on MacOS which had no files BTW, but mapping of other folder worked. It seems, mapping of special system folders is not working with UTM, but works fine in Parallels.

**Comment by johnnyshankman** [comment]
I cannot get this to work.

Made the Windows 10 VM with spice tools enabled etc. but my network folder is completely empty with nothing in it. There is no Network Drive anywhere to speak of.

I did install spice-tools when making the virt env, as I tried swapping from the other option, saving, then back to spice tools, and that did not fix the problem.

**Update: I was able to get it to work by reinstalling Guest Tools from `D:` on the VM's volumes, then shutting down the VM (which glitched out), _then_ restarting the entire UTM app from scratch, and _then_ starting up my VM once again.**

![Screenshot 2024-08-24 at 4 35 08 PM](https://github.com/user-attachments/assets/f846941b-7cdf-4268-b392-4bb8d8acb8e4)

**Reply by johnnyshankman** [reply]
Huzzah! Wasn't super easy but I got there.

When the guest tools finished installing my cursor stopped working in the VM, could not click on anything. The VM got real weird and corrupted. So I had to manually shut it down but then couldn't restart it.

I then shut the whole UTM program down etc, booted from clean, *now* Network Drive appears. 

![Screenshot 2024-08-24 at 4 49 40 PM](https://github.com/user-attachments/assets/8095b1a7-6017-4d43-8a7f-983c104ddbe0)

**Comment by maurice-schleussinger** [comment]
For me, an update to the latest release (UTM v4.5.4) fixed the issue. 🎉

**Reply by dis0908** [reply]
The issue doesn't seem to be fixed. I am using `UTM Version 4.7.5`. 

Steps:
- Installed `Guest tools`.
- Enabled `SharedFolder` and selected a directory to be shared with the host from guest.
- Checked in `/Volumes/*` and I didn't find it.

**Reply by johnnyshankman** [reply]
Did you try my debug instructions?

https://github.com/utmapp/UTM/discussions/3583#discussioncomment-10439735
